### PR TITLE
Add a check flag job with token.

### PR DESCRIPTION
**Pull request by zackgilbert**

### CODE REVIEW

File: app/models/check_flag_job.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    flag.save
+  end
+
+  def error(job, exception)
+    @exception = exception
+  end
+
+  def failure(job)
+    flag.job_id = nil
+    flag.status = "failed"
```
Comment (4sweep) [inline app/models/check_flag_job.rb]: i think it's best not to update job_id or status for the CheckFlagJob. if a check job failed, it's best to just leave its status as 'submitted' and only reflect the status/job id of the submission job. you can probably simply remove the success() and failure() methods here and stick with the default ones.

File: app/models/check_flag_job.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        rate_limit = 500
+      else
+        flags_of_type = flag.user.flags.where(:status => "queued").where("type NOT IN (?)", ["PhotoFlag", "TipFlag"]).count
+        rate_limit = 5000
+      end
+      # If this fails due to rate_limit_exceeded, figure out how long it will take to process all flags
+      # of this type and randomly assign a time in that window. It's hacky and brittle, but better than
+      # every job retrying at the same time.
+      # TODO: make this reschedule all likely to fail flags, not just this one
+      # TODO: create CheckFlagJob so that similar logic could be applied to that
```
Comment (4sweep) [inline app/models/check_flag_job.rb]: i think we can remove this comment from here and SubmitFlagJob

File: app/models/check_flag_job.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    # 12 gives last retry at 6 hours
+    return 12
+  end
+
+  def reschedule_at(time_now, attempts)
+    if (@exception.is_a?(Foursquare2::APIError) && @exception.type == 'rate_limit_exceeded') # Rate limit hit
+
+      # Photo and Tip flags have their own endpoint with their own rate limit
+      if ["PhotoFlag", "TipFlag"].include? flag.type
+        flags_of_type = flag.user.flags.where(:status => "queued").where(:type => flag.type).count
```
Comment (4sweep) [inline app/models/check_flag_job.rb]: Flags being checked are in the submitted stage, but we probably want to calculate this based on both queued and submitted. consider replacing the status line with:

:status => ["queued", "submitted"]

(here and on L37)

File: app/models/check_flag_job.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  def failure(job)
+    flag.job_id = nil
+    flag.status = "failed"
+    flag.save
+  end
+
+  def max_attempts
+    # Retries happen at 5 + n^4 seconds, where n = number of attempts
+    # 12 gives last retry at 6 hours
```
Comment (4sweep) [inline app/models/check_flag_job.rb]: this comment is out of date, since reschedule_at() overrides the logic. consider removing.

**Comment by zackgilbert** [comment]
@4sweep Thanks for the feedback! Those should all be resolved. I'll try to get this rolled out early Monday and keep an eye on things.

**Pull request merged**